presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -85,6 +96,26 @@ public RuleAssert on(Function<PlanBuilder, PlanNode> planProvider)
         return this;
     }
 
+    public RuleAssert withStats(Map<PlanNodeId, PlanNodeStatsEstimate> stats)
+    {
+        setStats(plan, stats);
+        return this;
+    }
+
+    private void setStats(PlanNode node, Map<PlanNodeId, PlanNodeStatsEstimate> stats)
+    {
+        if (stats.containsKey(node.getId())) {
```

Comment:
I would prefer `TestingLookup` to be build with stats for all nodes (e.g: using helper recursive method) instead of recreating `TestingLookup` for each node. I think the code would be cleaner then